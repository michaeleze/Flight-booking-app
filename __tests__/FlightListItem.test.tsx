import { act, cleanup, fireEvent, render, RenderResult, screen } from '@testing-library/react';
import FlightList from '@/components/FlightListItem';

const ITEM = {
    origin: 'LOS',
    departuredate: '10.10.2010',
    destination: 'FRA',
    returndate: '10.12.2010',
    seatavailability: '2',
    offertype: 'economy',
    price: {
      amount: '50',
      currency: 'EUR'
    }
};

describe('DateOption', () => {
  let component;

  beforeEach(() => {
    act(() => {
      component = render(
        <FlightList
          item={ITEM}
        />);
    });
  });

  afterEach(cleanup);

  it('should renders an item in component', async() => {
    const { getByText } = await component;
    const text = getByText(ITEM.departuredate);

    expect(text).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
