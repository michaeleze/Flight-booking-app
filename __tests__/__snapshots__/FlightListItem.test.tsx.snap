// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateOption should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="my-8 mx-auto flex flow-root max-w-5xl space-x-4 divide-y divide-gray-200 rounded-xl bg-white p-6 shadow-md dark:divide-gray-700 sm:flex-row md:flex-col"
        role="list"
      >
        <li
          class="container py-3 sm:py-4"
        >
          <div
            class="container flex flex-col items-center gap-y-6 space-x-4 md:flex-row md:gap-y-0"
          >
            <div
              class="flex min-w-0 flex-1 flex-col gap-6 text-center md:flex-row md:text-left"
            >
              <svg
                class="m-auto h-8 w-8 rounded-full "
                fill="currentColor"
                height="28"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="28"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M2.5 19h19v2h-19v-2zm19.57-9.36c-.21-.8-1.04-1.28-1.84-1.06L14.92 10l-6.9-6.43-1.93.51 4.14 7.17-4.97 1.33-1.97-1.54-1.45.39 2.59 4.49s7.12-1.9 16.57-4.43c.81-.23 1.28-1.05 1.07-1.85z"
                />
              </svg>
              <div
                class="flex flex-col gap-2"
              >
                <p
                  class="truncate text-sm font-medium text-gray-900 dark:text-white"
                >
                  LOS
                </p>
                <p
                  class="truncate text-sm text-gray-500 dark:text-gray-400"
                >
                  10.10.2010
                </p>
              </div>
            </div>
            <div
              class="flex min-w-0 flex-1 flex-col gap-6 text-center md:flex-row md:text-left"
            >
              <svg
                class=" m-auto h-8 w-8 rounded-full"
                fill="currentColor"
                height="28"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="28"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M2.5 19h19v2h-19v-2zm16.84-3.15c.8.21 1.62-.26 1.84-1.06.21-.8-.26-1.62-1.06-1.84l-5.31-1.42-2.76-9.02L10.12 2v8.28L5.15 8.95l-.93-2.32-1.45-.39v5.17l16.57 4.44z"
                />
              </svg>
              <div
                class="flex flex-col gap-2"
              >
                <p
                  class="truncate text-sm font-medium text-gray-900 dark:text-white"
                >
                  FRA
                </p>
                <p
                  class="truncate text-sm text-gray-500 dark:text-gray-400"
                >
                  10.12.2010
                </p>
              </div>
            </div>
            <div
              class="flex min-w-0 flex-1 gap-6"
            >
              <div
                class="flex flex-col gap-2"
              >
                <p
                  class="truncate text-center text-sm font-medium text-gray-900 dark:text-white"
                >
                  2
                </p>
                <p
                  class="truncate text-sm text-gray-500 dark:text-gray-400"
                >
                  Seat availability
                </p>
              </div>
            </div>
            <div
              class="flex min-w-0 flex-1 gap-6"
            >
              <div
                class="flex flex-col gap-2"
              >
                <p
                  class="truncate text-sm font-medium text-gray-900 dark:text-white"
                >
                  economy
                </p>
                <p
                  class="truncate text-sm text-gray-500 dark:text-gray-400"
                >
                  Offer type
                </p>
              </div>
            </div>
            <div
              class="inline-flex items-center text-base font-semibold text-gray-900 dark:text-white"
            >
              50
              EUR
            </div>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="my-8 mx-auto flex flow-root max-w-5xl space-x-4 divide-y divide-gray-200 rounded-xl bg-white p-6 shadow-md dark:divide-gray-700 sm:flex-row md:flex-col"
      role="list"
    >
      <li
        class="container py-3 sm:py-4"
      >
        <div
          class="container flex flex-col items-center gap-y-6 space-x-4 md:flex-row md:gap-y-0"
        >
          <div
            class="flex min-w-0 flex-1 flex-col gap-6 text-center md:flex-row md:text-left"
          >
            <svg
              class="m-auto h-8 w-8 rounded-full "
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M2.5 19h19v2h-19v-2zm19.57-9.36c-.21-.8-1.04-1.28-1.84-1.06L14.92 10l-6.9-6.43-1.93.51 4.14 7.17-4.97 1.33-1.97-1.54-1.45.39 2.59 4.49s7.12-1.9 16.57-4.43c.81-.23 1.28-1.05 1.07-1.85z"
              />
            </svg>
            <div
              class="flex flex-col gap-2"
            >
              <p
                class="truncate text-sm font-medium text-gray-900 dark:text-white"
              >
                LOS
              </p>
              <p
                class="truncate text-sm text-gray-500 dark:text-gray-400"
              >
                10.10.2010
              </p>
            </div>
          </div>
          <div
            class="flex min-w-0 flex-1 flex-col gap-6 text-center md:flex-row md:text-left"
          >
            <svg
              class=" m-auto h-8 w-8 rounded-full"
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M2.5 19h19v2h-19v-2zm16.84-3.15c.8.21 1.62-.26 1.84-1.06.21-.8-.26-1.62-1.06-1.84l-5.31-1.42-2.76-9.02L10.12 2v8.28L5.15 8.95l-.93-2.32-1.45-.39v5.17l16.57 4.44z"
              />
            </svg>
            <div
              class="flex flex-col gap-2"
            >
              <p
                class="truncate text-sm font-medium text-gray-900 dark:text-white"
              >
                FRA
              </p>
              <p
                class="truncate text-sm text-gray-500 dark:text-gray-400"
              >
                10.12.2010
              </p>
            </div>
          </div>
          <div
            class="flex min-w-0 flex-1 gap-6"
          >
            <div
              class="flex flex-col gap-2"
            >
              <p
                class="truncate text-center text-sm font-medium text-gray-900 dark:text-white"
              >
                2
              </p>
              <p
                class="truncate text-sm text-gray-500 dark:text-gray-400"
              >
                Seat availability
              </p>
            </div>
          </div>
          <div
            class="flex min-w-0 flex-1 gap-6"
          >
            <div
              class="flex flex-col gap-2"
            >
              <p
                class="truncate text-sm font-medium text-gray-900 dark:text-white"
              >
                economy
              </p>
              <p
                class="truncate text-sm text-gray-500 dark:text-gray-400"
              >
                Offer type
              </p>
            </div>
          </div>
          <div
            class="inline-flex items-center text-base font-semibold text-gray-900 dark:text-white"
          >
            50
            EUR
          </div>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
